sys/include/usb/usbus/msc/scsi.h

```diff
@@ -0,0 +1,293 @@
+#define SCSI_WRITE_AND_VERIFY           0x2E    /**< SCSI Write and Verify */
+#define SCSI_VERIFY10                   0x2F    /**< SCSI Verify10 */
+#define SCSI_MODE_SELECT10              0x55    /**< SCSI Mode Select10 */
+#define SCSI_MODE_SENSE10               0x5A    /**< SCSI Mode Sense10 */
+/** @} */
+
+/**
+ * @brief Command Block Wrapper signature
+ */
+#define SCSI_CBW_SIGNATURE              0x43425355
+
+/**
+ * @brief Command Status Wrapper signature
+ */
+#define SCSI_CSW_SIGNATURE              0x53425355
+
+/**
+ * @name USB SCSI Version list
+ * @{
+ */
+#define SCSI_VERSION_NONE               0x0000
+#define SCSI_VERSION_SCSI1              0x0001
+#define SCSI_VERSION_SCSI2              0x0002
+/** @} */
+
+/**
+ * @name USB SCSI Read format capacities descriptor type
+ *
+ * @see ( @ref msc_read_fmt_capa_pkt_t )
+ * @{
+ */
+#define SCSI_READ_FMT_CAPA_TYPE_RESERVED    0x00
+#define SCSI_READ_FMT_CAPA_TYPE_UNFORMATTED 0x01
+#define SCSI_READ_FMT_CAPA_TYPE_FORMATTED   0x02
+#define SCSI_READ_FMT_CAPA_TYPE_NO_MEDIA    0x03
+/** @} */
+
+/**
+ * @brief SCSI Request Sense error type
+ */
+#define SCSI_REQUEST_SENSE_ERROR              0x70
+
+/**
+ * @name USB SCSI Request Sense Sense Key type
+ *
+ * @{
+ */
+#define SCSI_SENSE_KEY_NO_SENSE             0x00
+#define SCSI_SENSE_KEY_RECOVERED_ERROR      0x01
+#define SCSI_SENSE_KEY_NOT_READY            0x02
+#define SCSI_SENSE_KEY_MEDIUM_ERROR         0x03
+#define SCSI_SENSE_KEY_HARDWARE_ERROR       0x04
+#define SCSI_SENSE_KEY_ILLEGAL_REQUEST      0x05
+#define SCSI_SENSE_KEY_UNIT_ATTENTION       0x06
+#define SCSI_SENSE_KEY_DATA_PROTECT         0x07
+#define SCSI_SENSE_KEY_BLANK_CHECK          0x08
+#define SCSI_SENSE_KEY_VENDOR_SPECIFIC      0x09
+#define SCSI_SENSE_KEY_ABORTED_COMMAND      0x0B
+#define SCSI_SENSE_KEY_VOLUME_OVERFLOW      0x0D
+#define SCSI_SENSE_KEY_MISCOMPARE           0x0E
+/** @} */
+
+/**
+ * @brief Packet structure to answer (@ref SCSI_TEST_UNIT_READY) request
+ *
+ * @see Inquiry Command from SCSI Primary Command
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;           /**< type of device */
+    uint8_t logical_unit;   /**< Selected LUN */
+    uint8_t reserved[10];   /**< Reserved */
+} msc_test_unit_pkt_t;
+
+/**
+ * @brief Packet structure to answer (@ref SCSI_MODE_SELECT6) and
+ * (@ref SCSI_MODE_SENSE6) requests
+ *
+ * @see Inquiry Command from SCSI Primary Command
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t mode_data_len;      /**< Size of the whole packet */
+    uint8_t medium_type;        /**< Type of medium */
+    uint8_t flags;              /**< Miscellaneous flags */
+    uint8_t block_desc_len;     /**< Length of block descriptor */
+} msc_mode_parameter_pkt_t;
+
+/**
+ * @brief CBW Packet structure for (@ref SCSI_READ10) and
+ * (@ref SCSI_WRITE10) requests
+ *
+ * @see Read10 or Write10 Command from SCSI Primary Command
+ *
+ * @warning uint16_t and uint32_t will be BIG ENDIAN
```

Comment:
better use `network_uint16_t`/`network_uint32_t` then

Reply:
Good idea !
But the `network_*` label puzzle me a bit.
How about `be_uint16_t` and so instead ?

Reply:
> How about `be_uint16_t` and so instead?

uh I didn't know we had both - yea `network_uint16_t` is just a typedef of `be_uint16_t` anyway, I just had never seen the latter.

Reply:
I've switched to `be_uint*_t` where applicable. I'll run some tests later this week to ensure I didn't break something.